widgetssdk/src/main/java/com/glia/widgets/entrywidget/EntryWidgetController.kt

```diff
@@ -1,20 +1,40 @@
 package com.glia.widgets.entrywidget
 
-internal class EntryWidgetController : EntryWidgetContract.Controller {
+import android.annotation.SuppressLint
+import com.glia.androidsdk.Engagement
+import com.glia.androidsdk.queuing.QueueState
+import com.glia.widgets.core.queue.QueueMonitor
+
+internal class EntryWidgetController(private val queueMonitor: QueueMonitor) : EntryWidgetContract.Controller {
     private lateinit var view: EntryWidgetContract.View
 
+    @SuppressLint("CheckResult")
     override fun setView(view: EntryWidgetContract.View) {
         this.view = view
-
-        // TODO: Remove this hardcoded list of items
-        val items = listOf(
-            EntryWidgetContract.ItemType.VIDEO_CALL,
-            EntryWidgetContract.ItemType.AUDIO_CALL,
-            EntryWidgetContract.ItemType.CHAT,
-            EntryWidgetContract.ItemType.SECURE_MESSAGE,
-            EntryWidgetContract.ItemType.PROVIDED_BY
+        val items = mutableListOf(
+            EntryWidgetContract.ItemType.EMPTY_STATE
         )
         view.showItems(items)
+
+        queueMonitor.observableIntegratorQueues.subscribe {
+            // TODO: code below should be moved to a standalone component
+            val openQueues = it.filter { queueFromList -> queueFromList.state.status == QueueState.Status.OPEN }
+            val allMedias = openQueues.flatMap { queueFromList -> queueFromList.state.medias.toList() }.distinct()
+
+            items.clear()
```

Comment:
I'd prefer to have immutable list here.

Reply:
Changed: f91942e868436ea28597504e3817387ba1e0bb5c. Please have a look to see if it's what you suggested.

Reply:
sorry for the poorly described request.
I meant not to keep the mutable list of items and clear it each time and then add new items.

I meant smth like this
```kotlin
override fun setView(view: EntryWidgetContract.View) {
        this.view = view
        view.showItems(listOf(EntryWidgetContract.ItemType.EMPTY_STATE))

        queueMonitor.observableIntegratorQueues.subscribe {
            // TODO: code below should be moved to a standalone component
            val openQueues = it.filter { queueFromList -> queueFromList.state.status == QueueState.Status.OPEN }
            val allMedias = openQueues.flatMap { queueFromList -> queueFromList.state.medias.toList() }.distinct()

            val items = buildList {
                if (allMedias.contains(Engagement.MediaType.TEXT)) add(EntryWidgetContract.ItemType.CHAT)
                if (allMedias.contains(Engagement.MediaType.AUDIO)) add(EntryWidgetContract.ItemType.AUDIO_CALL)
                if (allMedias.contains(Engagement.MediaType.VIDEO)) add(EntryWidgetContract.ItemType.VIDEO_CALL)
                if (allMedias.contains(Engagement.MediaType.MESSAGING)) add(EntryWidgetContract.ItemType.SECURE_MESSAGE)

                if (allMedias.isEmpty()) {
                    add(EntryWidgetContract.ItemType.EMPTY_STATE)
                } else {
                    add(EntryWidgetContract.ItemType.PROVIDED_BY)
                }
            }
            view.showItems(items)
        }
    }
``` 

Or maybe one more way is to use Rx advantages like this
```kotlin
    override fun setView(view: EntryWidgetContract.View) {
        this.view = view
        view.showItems(listOf(EntryWidgetContract.ItemType.EMPTY_STATE))

        // TODO: code below should be moved to a standalone component
        queueMonitor.observableIntegratorQueues
            .map { queues -> queues.map { it.state } }
            .map { states -> states.filter { it.status == QueueState.Status.OPEN } }
            .map { openQueues -> openQueues.flatMap { it.medias.toList() }.distinct() }
            .map { allMedias ->
                buildList {
                    if (allMedias.contains(Engagement.MediaType.TEXT)) add(EntryWidgetContract.ItemType.CHAT)
                    if (allMedias.contains(Engagement.MediaType.AUDIO)) add(EntryWidgetContract.ItemType.AUDIO_CALL)
                    if (allMedias.contains(Engagement.MediaType.VIDEO)) add(EntryWidgetContract.ItemType.VIDEO_CALL)
                    if (allMedias.contains(Engagement.MediaType.MESSAGING)) add(EntryWidgetContract.ItemType.SECURE_MESSAGE)

                    if (allMedias.isEmpty()) {
                        add(EntryWidgetContract.ItemType.EMPTY_STATE)
                    } else {
                        add(EntryWidgetContract.ItemType.PROVIDED_BY)
                    }
                }
            }
            //This way we can avoid switching to main thread in Adapter
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe(view::showItems)
    }
```

Reply:
Ah, I understand now. Yes, the Rx approach is better because it also avoids switching to main thread in adapter. 👍 
Made changes here 5b65432c8abc8bf2b5d37939cca3fdbede9bbfa6. Thank you.